provider/marathon/marathon_test.go

```diff
@@ -580,14 +689,37 @@ func TestMarathonGetPort(t *testing.T) {
 			task:        createTask(taskPorts(7777)),
 			expected:    "7777",
 		},
+		{
+			desc:        "multiple task ports with service index available",
+			application: createApplication(label(types.LabelPrefix+"http.portIndex", "0")),
+			task:        createTask(taskPorts(80, 443)),
+			expected:    "80",
+		},
+		{
+			desc:        "multiple task ports with service port available",
+			application: createApplication(label(types.LabelPrefix+"https.port", "443")),
+			task:        createTask(taskPorts(80, 443)),
+			expected:    "443",
+		},
+		{
+			desc:        "multiple task ports with services but default port available",
+			application: createApplication(label(types.LabelPrefix+"http.weight", "100")),
+			task:        createTask(taskPorts(80, 443)),
+			expected:    "80",
+		},
 	}
 
 	for _, c := range cases {
 		c := c
 		t.Run(c.desc, func(t *testing.T) {
 			t.Parallel()
-			actual := provider.getPort(c.task, c.application)
-			if actual != c.expected {
+			if actual, ok := assertWithServices(provider.getServiceNames(c.application), c.expected,
+				func(serviceName string) string {
+					if value := provider.getPort(c.task, c.application, serviceName); len(value) > 0 {
+						return value
+					}
+					return ""
+				}); !ok {
```

Comment:
The expected values are already in the test table, the reason for the function is to extract the common boiler-plate code to loop through the service names to get the right value...